### ISSUE TITLE
Generated markdown is not compatible with Test Summary Action

### ISSUE THREAD

**Issue by JavierSegoviaCordoba** [detekt/detekt#5102]
I think one of the main purposes for that report was using it on Action, not sure if this should be reported as a bug or not tho.

![image](https://user-images.githubusercontent.com/7463564/179698556-0e7302f4-8fb0-4a99-bd47-ecf3d6d4e8c7.png)

**Comment by cortinico** [comment]
Could you post the content of your report?

**Comment by JavierSegoviaCordoba** [comment]
I have searched on their repo and I found nothing :/ 

My md starts so:

```md
# detekt

## Metrics

* 128 number of properties

* 70 number of functions
```

**Comment by BraisGabin** [comment]
What should we do here? The markdown seems fine. Should this be reported to the action itself? Maybe it's a bug on its side and if it's not they will point out what's the problem in our markdown.

**Comment by JavierSegoviaCordoba** [comment]
I can't even found the log message in their source code 🫠

So we are totally blocked and there is no other way than asking them.

**Comment by JavierSegoviaCordoba** [comment]
https://github.com/test-summary/action/issues/10

**Comment by 3flex** [comment]
I googled the error message and it seems to be coming from a Node lib called xml2js, which was odd because you're feeding markdown to the action, not XML.

Then I looked at the action's README, which says it takes XML or TAP format and generates markdown output.

So unless I've misunderstood, the action doesn't take markdown input at all, so there's nothing for detekt to change here. It doesn't make sense to use detekt output as an input to that action.

@JavierSegoviaCordoba can you please confirm?

**Comment by JavierSegoviaCordoba** [comment]
You are totally right, I read the Detekt feature request about the Markdown report and thought I had to use a markdown file instead of an XML file.

https://github.com/detekt/detekt/issues/4826

**Comment by JavierSegoviaCordoba** [comment]
Additionally, the `detekt.xml` file is not compatible with the summary action.

```
Error: unknown test file type for 'build/reports/detekt/detekt.xml'
```

Maybe I can create a different issue

**Comment by cortinico** [comment]
I think you should instead use the `$GITHUB_STEP_SUMMARY` variable rather than using the test-action report: https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/#create-summaries
